### ISSUE TITLE
RedisCluster scan documentation and parameter order issue

### ISSUE THREAD

**Issue by kjoe** [phpredis/phpredis#1356]
Please improve RedisCluster documentation, because to use scan command, you must iterate over all the redis master nodes.

Parameter ordering of RedisCluster scan command differs from the other "Directed node commands":

https://github.com/phpredis/phpredis/blob/97bb6bd85e3fa38dbb13b97c50bd315490265a80/redis_cluster.c#L101-L106

https://github.com/phpredis/phpredis/blob/97bb6bd85e3fa38dbb13b97c50bd315490265a80/redis_cluster.c#L2569-L2570

Or it's just a mispelling/bug, as the header prototype shows the conventional (correct) order:

https://github.com/phpredis/phpredis/blob/97bb6bd85e3fa38dbb13b97c50bd315490265a80/redis_cluster.c#L2547-L2548

**Comment by yatsukhnenko** [comment]
I prefer to say that this is bug in comment :smile:
@michael-grunder could you look?

**Comment by xergio** [comment]
It seems the syntax is something like this:

```php
$r = new RedisCluster(null, ["localhost:6379"]);

foreach ($r->_masters() as $master) {
	$it = null;
	while ($keys = $r->scan($it, $master, "pattern...")) {
		foreach ($keys as $key) {
			// code
		}
	}
}
```

Comments and documentation should be fixed, in this case the `$master` (node) is the second argument.

**Comment by michael-grunder** [comment]
Should be fixed via fe21524e and 8838f534.  We probably want to add a note to all such commands.
